pcsx2/GameDatabase.cpp

```diff
@@ -14,63 +14,193 @@
+		if (i != memcardFilters.size() - 1)
+			filters.append(",");
+	}
+	return filters;
 }
 
-// Sets the current game to the one matching the serial id given
-// Returns true if game found, false if not found...
-bool BaseGameDatabaseImpl::findGame(Game_Data& dest, const wxString& id) {
+std::string YamlGameDatabaseImpl::safeGetString(const YAML::Node& n, std::string key, std::string def)
+{
+	if (!n[key])
+		return def;
+	return n[key].as<std::string>();
+}
 
-	GameDataHash::const_iterator iter( gHash.find(id) );
-	if( iter == gHash.end() ) {
-		dest.clear();
-		return false;
+int YamlGameDatabaseImpl::safeGetInt(const YAML::Node& n, std::string key, int def)
+{
+	if (!n[key])
+		return def;
+	return n[key].as<int>();
+}
+
+std::vector<std::string> YamlGameDatabaseImpl::safeGetMultilineString(const YAML::Node& n, std::string key, std::vector<std::string> def)
+{
+	if (!n[key])
+		return def;
+
+	std::vector<std::string> lines;
+
+    std::istringstream stream(safeGetString(n, key));
+	std::string line;
+
+	while(std::getline(stream, line)) {
+		lines.push_back(line);
 	}
-	dest = iter->second;
-	return true;
+
+	return lines;
 }
 
-Game_Data* BaseGameDatabaseImpl::createNewGame( const wxString& id )
+std::vector<std::string> YamlGameDatabaseImpl::safeGetStringList(const YAML::Node& n, std::string key, std::vector<std::string> def)
 {
-	return &gHash.emplace(id, Game_Data{id}).first->second;
+	if (!n[key])
+		return def;
+	return n[key].as<std::vector<std::string>>();
 }
 
-// Searches the current game's data to see if the given key exists
-bool Game_Data::keyExists(const wxString& key) const {
-	for (auto it = kList.begin(); it != kList.end(); ++it) {
-		if (it->CompareKey(key)) {
-			return true;
+GameDatabaseSchema::GameEntry YamlGameDatabaseImpl::entryFromYaml(const std::string serial, const YAML::Node& node)
+{
+	GameDatabaseSchema::GameEntry gameEntry;
+	try
+	{
+		gameEntry.name = safeGetString(node, "name");
+		gameEntry.region = safeGetString(node, "region");
+		gameEntry.compat = static_cast<GameDatabaseSchema::Compatibility>(safeGetInt(node, "compat", enum_cast(gameEntry.compat)));
+		gameEntry.eeRoundMode = static_cast<GameDatabaseSchema::RoundMode>(safeGetInt(node, "eeRoundMode", enum_cast(gameEntry.eeRoundMode)));
+		gameEntry.vuRoundMode = static_cast<GameDatabaseSchema::RoundMode>(safeGetInt(node, "vuRoundMode", enum_cast(gameEntry.vuRoundMode)));
+		gameEntry.eeClampMode = static_cast<GameDatabaseSchema::ClampMode>(safeGetInt(node, "eeClampMode", enum_cast(gameEntry.eeClampMode)));
+		gameEntry.vuClampMode = static_cast<GameDatabaseSchema::ClampMode>(safeGetInt(node, "vuClampMode", enum_cast(gameEntry.vuClampMode)));
+
+		// Validate game fixes, invalid ones will be dropped!
+		for (std::string fix : safeGetStringList(node, "gameFixes"))
+		{
+			bool fixValidated = false;
+			for (GamefixId id = GamefixId_FIRST; id < pxEnumEnd; ++id)
+			{
+				std::string validFix = wxString(EnumToString(id)).ToStdString() + L"Hack";
```

Comment:
Doesn't compile. Let's do this:
std::string validFix = wxString(EnumToString(id)).Append(L"Hack").ToStdString();